lib/matplotlib/rcsetup.py

```diff
@@ -718,6 +719,54 @@ def visit_Attribute(self, node):
         self.generic_visit(node)
 
 
+def validate_legend_loc(loc):
+    """
+    Confirm that loc is a type which rc.Params["legend.loc"] supports.
+
+    .. versionchanged:: 3.7
+
+    Parameters
+    ----------
+    loc : str | int | (float, float) | str((float, float))
+        The location of the legend.
+
+    Returns
+    -------
+    loc : str | int | (float, float) or raise ValueError exception
+        The location of the legend.
+    """
+    valid = {
+        "best",
+        "upper right", "upper left", "lower left", "lower right", "right",
+        "center left", "center right", "lower center", "upper center",
+        "center"
+    }
+
+    key = "legend.loc"
+    msg = (f"{loc!r} is not a valid value for {key}; supported values "
+           f"are {', '.join(valid)} | tuple[float, float] "
+           f"| str(tuple[float, float])")
+
+    if isinstance(loc, int):
+        return loc
```

Comment:
I think the int handling should go below the string handling:  if the user sets the int in their matplotlibrc, similar to the tuple case, it would need converting from string to int before checking.